com.unity.sg2/Tests/GraphUI/DataModel/Constants/TextureTypeConstantTest.cs

```diff
@@ -0,0 +1,76 @@
+using NUnit.Framework;
+using Unity.GraphToolsFoundation;
+using UnityEditor.ShaderGraph.Defs;
+using UnityEditor.ShaderGraph.GraphDelta;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+
+namespace UnityEditor.ShaderGraph.GraphUI.UnitTests.DataModel.Constants
+{
+    class TextureTypeConstantTest : BaseGraphAssetTest
+    {
+        static TestCaseData[] s_FieldTypesAndSampleValues =
+        {
+            new TestCaseData(TYPE.Texture2D, new Texture2D(1, 1)).SetName("{m}(Texture2D)"),
+            new TestCaseData(TYPE.Texture3D, new Texture3D(1, 1, 1, default(DefaultFormat), default)).SetName("{m}(Texture3D)"),
+            new TestCaseData(TYPE.TextureCube, new Cubemap(1, default(DefaultFormat), default)).SetName("{m}(Cubemap)"),
+            new TestCaseData(TYPE.Texture2DArray, new Texture2DArray(1, 1, 1, default(DefaultFormat), default)).SetName("{m}(Texture2DArray)"),
+        };
+
+        static TestCaseData[] s_FieldTypesAndTypeHandles =
+        {
+            new TestCaseData(TYPE.Texture2D, ShaderGraphExampleTypes.Texture2DTypeHandle).SetName("{m}(Texture2D)"),
+            new TestCaseData(TYPE.Texture3D, ShaderGraphExampleTypes.Texture3DTypeHandle).SetName("{m}(Texture3D)"),
+            new TestCaseData(TYPE.TextureCube, ShaderGraphExampleTypes.CubemapTypeHandle).SetName("{m}(Cubemap)"),
+            new TestCaseData(TYPE.Texture2DArray, ShaderGraphExampleTypes.Texture2DArrayTypeHandle).SetName("{m}(Texture2DArray)"),
+        };
+
+        [Test]
+        [TestCaseSource(nameof(s_FieldTypesAndSampleValues))]
+        public void TestGetObjectValue_MatchesField(ITypeDescriptor type, Texture value)
```

Comment:
For anything to do with a type that can store a reference to assets like texture types, it might be nice to have a set of tests for actual texture assets as well? and then do an asset data base check if its valid or something